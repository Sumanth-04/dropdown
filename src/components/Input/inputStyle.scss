@import '../variables';


.rd-input::placeholder, .rd-textarea::placeholder {
  opacity: 0.7;
  font-size:$placeHolder-font-size;
}
.rd-input , .rd-textarea {
  border: 1px solid $primaryBorder-color;
  padding-left: 8px;
  border-radius: 0px;
  height: $inputFieldHeight;
  &:hover{
    background : #F0F8FF;
  }
}
.rd-textarea {
  height: $text-areaFieldHeight;
}



