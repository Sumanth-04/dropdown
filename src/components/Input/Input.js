import React, {Component} from 'react';
import PropTypes from 'prop-types';

import './inputStyle.scss';

class Input extends Component{
	static propTypes = {
		label : PropTypes.element,
		inputStyles: PropTypes.object,
		className: PropTypes.string
	};
	
	constructor(props){
		super(props);
	}

	render(){
		const {
			label
		} = this.props;

		return (
			<div>
				{label ? <span className="rd-label">{label}</span> : null}
				<input className={`rd-input  ${this.props.className}`} style={{...this.props.inputStyles}} {...this.props}  />
			</div>
		)
	}
}

export default Input;