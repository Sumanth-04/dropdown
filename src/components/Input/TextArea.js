import React, {Component} from 'react';
import Textarea from 'rc-textarea';
import PropTypes from 'prop-types';

import './inputStyle.scss';

class TextArea extends Component{
	static propTypes = {
		textAreaStyles: PropTypes.object,
		className: PropTypes.string,
	};
	
	constructor(props){
		super(props);
	}

	render(){
		return (
			<Textarea className={`rd-textarea ${this.props.className}`} {...this.props} style={{...this.props.textAreaStyles}} />
		)
	}
}

export default TextArea;