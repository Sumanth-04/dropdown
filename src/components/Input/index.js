import Input from './Input';
import TextArea from './TextArea';

Input.Input = Input;
Input.TextArea = TextArea;

export default Input;