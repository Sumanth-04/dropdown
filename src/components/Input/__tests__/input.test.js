import React from "react";
import renderer from "react-test-renderer";
import Input from "../Input";
import TextArea from "../index";
//import {isTSAnyKeyword} from '@babel/types';

it("Input should have a label", () => {
  const tree = renderer
    .create(
      <Input
        label={<span>Enter Your Name</span>}
        placeholder="First Name"
      ></Input>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("TextArea should have a label", () => {
  const tree = renderer
    .create(
      <Input.TextArea textAreaStyles={{}} placeholder="First Name"></Input.TextArea>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});