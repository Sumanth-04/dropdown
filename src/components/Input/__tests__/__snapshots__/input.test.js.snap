// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should have a label 1`] = `
<div>
  <span
    className="rd-label"
  >
    <span>
      Enter Your Name
    </span>
  </span>
  <input
    className="rd-input  undefined"
    label={
      <span>
        Enter Your Name
      </span>
    }
    placeholder="First Name"
    style={Object {}}
  />
</div>
`;

exports[`Input should have a label 2`] = `
<div>
  <input
    className="rd-input  undefined"
    style={Object {}}
  />
</div>
`;

exports[`Input should have a label 3`] = `
<textarea
  className="rc-textarea rd-textarea undefined"
  onChange={[Function]}
  onKeyDown={[Function]}
  style={Object {}}
  value=""
/>
`;

exports[`TextArea should have a label 1`] = `
<textarea
  className="rc-textarea rd-textarea undefined"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="First Name"
  style={Object {}}
  textAreaStyles={Object {}}
  value=""
/>
`;
